include/openPMD/Iteration.hpp

```diff
@@ -97,7 +111,21 @@ class Iteration : public Attributable
     void flushFileBased(std::string const&, uint64_t);
     void flushGroupBased(uint64_t);
     void flush();
-    void read();
+    void
+    read();
+
+    /*
+     * Indicates that an iteration has been logically closed.
+     * Will be physically closed upon next flush.
+     */
+    std::shared_ptr< bool > m_closed = std::make_shared< bool >( false );
+
+    /*
+     * Once an iteration is declared finalized, we do not flush it any longer
+     * to avoid reading undefined data if a backend implements optimizations
+     * based on this information.
+     */
+    std::shared_ptr< bool > skipFlush = std::make_shared< bool >( false );
```

Comment:
why do we need both `m_closed` and `[m_]skipFlush`?

Reply:
Documentation here is outdated, will fix – also maybe the naming should be redone, something like `logicallyClosed` and `physicallyClosed`. `logicallyClosed` means that the iteration has been closed, but the corresponding `flush` has not run yet.

Reply:
then I would just call it `m_closed` and `m_flushed` or so. (we have no control when something really is on disk, that is up to the backend we talk to.)

Reply:
Hm, I think `m_flushed` does not sound very clear about what it's doing?
I mean, while you can certainly argue that `m_physicallyClosed` is not *technically* correct, it is "physically closed" from the frontend's perspective. Anything from now on is the backend's problem which the frontend needs not and cannot interfere with any longer. I'd say that's close enough? :D 

Alternatively, what about `m_closedInFrontend` and `m_closedInBackend`?

Reply:
Changed this thing into an enumeration that can take the values `Open`, `ClosedInFrontend` and `ClosedInBackend` now.